### DISCUSSION TITLE
Does retryFits optimize the model fit?

### DISCUSSION THREAD

**Discussion by mebowers5** [bmcclintock/momentuHMM#102] [Q&A] [answered]
I am using the momentuHMM package function crawlWrap to "connect the dots" between some discrete movement data. I have a question regarding the difference between the attempts and retryFits arguments. I understand that attempts is the number of times the likelihood optimization will attempt to converge the fit in the case that convergence is not reached but, retryFits will proceed to do the same thing regardless of whether convergence was achieved. My question is this: is retryFits optimizing the fit in any way? In other words, if I set attempts to 15 and retryFits to zero, am I preventing some additional optimization of the fit by doing so? I am running 1,000 iterations for my analysis and the retryFits option seems to add a lot of processing time and memory so I would like to omit it if it makes no difference.

**Comment by bmcclintock** [comment, marked as answer]
`retryFits` does indeed re-optimize (or re-fit) the model using random perturbations of the current estimates as the initial values. The degree of perturbation is determined by the `retrySD` argument. This can be helpful for dealing with "false" convergence by exploring the parameter space. Conversely, `attempts` is used within `crawl::crwMLE` and it does not use random perturbation; it's more meant to deal with convergence errors and will yield identical estimates unless the `initialSANN` argument is specified (in which case the simulated annealing can often yield different initial values for the optimization). So it does make a difference, particularly if the likelihood is relatively flat (and if the likelihood is relatively flat, it might be worth specifying priors for the parameters via the `priors` argument).

One strategy would be setting `attempts` to a large number (as it will stop once the initial fit doesn't produce an error) and setting `retryFits` to the lowest reasonable value. You can get a sense for how stable the likelihood optimization is by watching the printed output during model fits; if the negative log-likelihood continues to decrease as the number of re-fits increases, this would indicate the optimization is unstable and warrants a large number for `retryFits`.  See #79 for some more info and tips.
